packages/core/src/browser/menu/browser-menu-plugin.ts

```diff
@@ -324,22 +335,6 @@ export class DynamicMenuWidget extends MenuWidget {
         return false;
     }
 
-    protected runWithPreservedFocusContext(what: () => void): void {
```

Comment:
@kittaakos do you remember why we had it?

Reply:
Sure, we had to snapshot the command handlers' visibility/enablement before setting the focus on the menu node. Otherwise, certain handlers that rely on the selection did not work correctly.

Reply:
https://github.com/eclipse-theia/theia/pull/7481

Reply:
@marechal-p Did your test cover it?

Reply:
@kittaakos what is the link with https://github.com/eclipse-theia/theia/pull/7481 ? I can't find the removed code in it.

@akosyakov I did not specifically test what @kittaakos described, how should one test that?

Reply:
> @kittaakos what is the link with #7481 ? I can't find the removed code in it.

Here is the commit: https://github.com/eclipse-theia/theia/pull/7481/commits/e33168e35b3e50c825c24d6d2ddce49e6a7208c9

Reply:
Found it thanks! Reading the code before and after these changes, commands are still snapshot. The only difference is that we don't use `runWithPreservedFocusContext`. How can we test for regressions in this case?